src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs

```diff
@@ -138,9 +138,7 @@ public class EventHubDataComparer : ICacheDataComparer<CachedEventHubMessage>
         public int Compare(CachedEventHubMessage cachedMessage, StreamSequenceToken streamToken)
         {
             var realToken = (EventSequenceToken)streamToken;
-            return cachedMessage.SequenceNumber != realToken.SequenceNumber
-                ? (int)(cachedMessage.SequenceNumber - realToken.SequenceNumber)
-                : 0 - realToken.EventIndex;
+            return (int)(cachedMessage.SequenceNumber - realToken.SequenceNumber);
```

Comment:
No, and the previous logic contained an extremely rare bug.  The event index is index of an item in a batch, and the cache only deals with full batches.  So by checking the index, we would not recognize that this is in fact the right batch but instead go to the previous batch.  Normally this is fine as duplicates are allowed, however, in the rare case that this batch (the right batch) is the last item in the cache, we'd fail to find a batch even when one was available.  Absurdly rare edge case, but wrong none-the-less.